packages/api/tests/lib/rules/test-updateRuleResources.js

```diff
@@ -0,0 +1,475 @@
+      retries: 4,
+    },
+  });
+  const sqsRule = await createRuleTrigger(rule);
+  t.deepEqual(sqsRule.value, queues.queueUrl);
+
+  const newQueues = await createSqsQueues(randomString());
+  const updatedRule = {
+    ...sqsRule,
+    value: newQueues.queueUrl,
+  };
+  const updatedSqsRule = await updateRuleTrigger(sqsRule, updatedRule, t.context.testKnex);
+  t.deepEqual(updatedSqsRule.value, newQueues.queueUrl);
+  t.teardown(async () => {
+    await SQS.deleteQueue(queues.queueUrl);
+    await SQS.deleteQueue(newQueues.queueUrl);
+  });
+});
+
+test.serial('Updating an SNS rule updates the event source mapping', async (t) => {
+  const snsTopicArn = randomString();
+  const newSnsTopicArn = randomString();
+  const { TopicArn } = await awsServices.sns().createTopic({
+    Name: snsTopicArn,
+  }).promise();
+  const { TopicArn: TopicArn2 } = await awsServices.sns().createTopic({
+    Name: newSnsTopicArn,
+  }).promise();
+
+  const lambdaStub = sinon.stub(awsServices, 'lambda')
+    .returns({
+      addPermission: () => ({
+        promise: () => Promise.resolve(),
+      }),
+      removePermission: () => ({
+        promise: () => Promise.resolve(),
+      }),
+    });
+  const snsStub = sinon.stub(awsServices, 'sns')
+    .returns({
+      listSubscriptionsByTopic: () => ({
+        promise: () => Promise.resolve({
+          Subscriptions: [{
+            Endpoint: process.env.messageConsumer,
+            SubscriptionArn: randomString(),
+          }],
+        }),
+      }),
+      unsubscribe: () => ({
+        promise: () => Promise.resolve(),
+      }),
+    });
+
+  const unsubscribeSpy = sinon.spy(awsServices.sns(), 'unsubscribe');
+  const rule = fakeRuleRecordFactory({
+    type: 'sns',
+    value: TopicArn,
+    workflow,
+    enabled: true,
+  });
+
+  const ruleWithTrigger = await createRuleTrigger(rule);
+
+  t.is(rule.value, TopicArn);
+
+  const updates = {
+    ...rule,
+    arn: ruleWithTrigger.arn,
+    value: TopicArn2,
+  };
+  const updatedSnsRule = await updateRuleTrigger(ruleWithTrigger, updates, t.context.testKnex);
+  t.true(unsubscribeSpy.called);
+  t.true(unsubscribeSpy.calledWith({ SubscriptionArn: updates.arn }));
+
+  t.is(updatedSnsRule.name, rule.name);
+  t.is(updatedSnsRule.type, rule.type);
+  t.is(updatedSnsRule.value, TopicArn2);
+  t.not(updatedSnsRule.arn, rule.arn);
+
+  t.teardown(async () => {
+    lambdaStub.restore();
+    snsStub.restore();
+    unsubscribeSpy.restore();
+    await awsServices.sns().deleteTopic({ TopicArn: TopicArn2 }).promise();
+  });
+});
+
+test.serial('Enabling a disabled SNS rule and passing rule.arn throws specific error', async (t) => {
+  const snsTopicArn = randomString();
+  const snsStub = sinon.stub(awsServices, 'sns')
+    .returns({
+      listSubscriptionsByTopic: () => ({
+        promise: () => Promise.resolve({
+          Subscriptions: [{
+            Endpoint: process.env.messageConsumer,
+            SubscriptionArn: snsTopicArn,
```

Comment:
```suggestion
            SubscriptionArn: randomString(),
```

The subscription ARN for the Lambda subscription to the SNS topic is separate from the SNS topic ARN

Reply:
[`aa59c90` (#2871)](https://github.com/nasa/cumulus/pull/2871/commits/aa59c90a42ee176d4075b4de2c1bf78678177b09#diff-057c67806291c57c6410d819b57bda8d87b5370152a299e6baca44b5a44f0e40R436)